src/main/java/ah/cryptocoin/domain/ApplicationConf.java

```diff
@@ -38,33 +51,81 @@ public static ApplicationConf getApplicationConf() {
 
     private ApplicationConf() {
         conf = this;
-        compareMode = PriceComparison.PREVIOUS;
+        compareMode = PriceComparison.THRESHOLD;
         rateToCompare = new Rate(-1);
-        loadHistoricalRate();
+        loadCBMonitorData();
         Runtime.getRuntime().addShutdownHook(new Thread(){
             @Override
             public void run() {
                 shutdown();
             }
         });
+    }
 
+    private void checkAndCreateAppDataDir() {
+        File f = new File(CbMonitorHome);
+        if (!f.exists()) {f.mkdir();}
+    }
+    private void loadCBMonitorData() {
+        checkAndCreateAppDataDir();
+        loadRateHistory();
+        loadThresholdRate();
+        Agent.register(historicalRate);
     }
 
-    private void loadHistoricalRate() {
-        File f = new File(System.getProperty("user.home")+File.separator+"coinbase-rh.json");
-        if(!f.exists()) {
-            historicalRate = new RateHistory(new ArrayList<Rate>());
-        } else {
-            try {
-                InputStreamReader isr = new InputStreamReader(new FileInputStream(f));
-                Gson gson = new Gson();
-                historicalRate = gson.fromJson(isr, RateHistory.class);
-            } catch (FileNotFoundException e) {
-                e.printStackTrace();
-            }
+    private void loadRateHistory() {
+        try {
+            historicalRate = loadData(RateHistory.class, rateHistoryFile);
+        } catch (Exception e) {
+            historicalRate = new RateHistory();
+        }
+    }
+
+    private void loadThresholdRate() {
+        try {
+            thresholdRate = loadData(Rate.class, thresholdRateFile);
+        } catch (Exception e) {
+            thresholdRate = new Rate(System.currentTimeMillis());
+            thresholdRate.setBuy(createCoin(4000f, Coin.CoinBase.BTC, Coin.Currency.USD),
```

Comment:
What is need for setting it to some default values

Reply:
Default values are set because we haven't built preference yet and also we don't have a UI to change now thats why defaulted now.  Once we have a UI for setting Threshold we can popup that and get value from user before switching.

Currently since i have defaulted the comparison mode to Threshold i have to set some value without UI.  Prior to this implementation the comparison mode was set to Previous rate value fetched from API, it is not the case anymore from this PR.